src/libraries/Common/src/System/Security/Cryptography/ECDiffieHellmanCng.ImportExport.cs

```diff
@@ -35,8 +36,20 @@ public override void ImportParameters(ECParameters parameters)
                             SR.Format(SR.Cryptography_InvalidCurveOid, curve.Oid.Value));
                     }
 
-                    byte[] ecNamedCurveBlob = ECCng.GetNamedCurveBlob(ref parameters, ecdh: true);
-                    ImportKeyBlob(ecNamedCurveBlob, curve.Oid.FriendlyName, includePrivateParamerters);
+                    if (!hasPublicParameters && includePrivateParameters)
+                    {
+                        byte[] zero = new byte[parameters.D!.Length];
```

Comment:
We could instead change `GetNamedCurveBlob` to handle skipping over X and Y since it builds a zeroed blob anyway... but implementing this in `GetNamedCurveBlob` directly seemed like putting the support for this at the wrong place.